### ISSUE TITLE
False positive for extra semicolon

### ISSUE THREAD

**Issue by lzybkr** [intellij-rust/intellij-rust#5413]
## Environment

* **IntelliJ Rust plugin version:**
0.2.122.3105-201
* **Rust toolchain version:**
1.43.0
* **IDE name and version:**
Rider 2020.1.3
* **Operating system:**
Windows 10

## Problem description

There are several problem:

* the extension incorrectly detects an extra semicolon
* the highlighted text seems unrelated because it seems like the macro what's causing the trouble - replace the macro and the issue goes away
* applying the fix makes no changes but appears to work as the coloring reverts to normal until you edit anything

![image](https://user-images.githubusercontent.com/2148248/82525147-923ed100-9ae5-11ea-9498-7a1dd4f0a842.png)

## Steps to reproduce

This simple code reproduces the issue:

```rust
macro_rules! bail {
    () => {return Err()};
}

fn f() -> Result<(), ()> {
    if false
    {
        return Ok(());
    }

    bail!();
}

```

**Comment by mili-l** [comment]
see https://github.com/intellij-rust/intellij-rust/pull/9729
